### DISCUSSION TITLE
Opening WinForm XML Island with a SettingCard throws `XamlParseException: XAML parsing failed`

### DISCUSSION THREAD

**Discussion by karmeye** [CommunityToolkit/Windows#345] [Q&A]
Using a `SettingsCard` like this

```
<UserControl...
...
<ct:SettingsCard x:Name="settingsCard"
					   Description="{x:Bind vm:Config.Instance.ServerIP, Mode=OneWay}"
					   Header="Server Address"
					   HeaderIcon="{ui:FontIcon Glyph=}"
					   IsEnabled="true"
					   >
	<Button x:Name="button_CheckConnection"
			Content="{StaticResource App.String.Edit}"
			Click="Button_Click" />

</ct:SettingsCard>
...
</UserControl>
```

and opening it in a WinForms a via XML Island throws this exception:

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
Microsoft.UI.Xaml.Markup.XamlParseException: XAML parsing failed.
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
   at ABI.Microsoft.UI.Xaml.IApplicationStaticsMethods.LoadComponent(IObjectReference _obj, Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at Microsoft.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
```

- Calling the same XAML from a WinUI `Window` works. 
- The XAML above is contained in a WinUI Class Library UserControl which the WinForms executable project references as a Project reference. Just removing the `SettingsCard` works (if I copy a folder generated by buiding the WinUI Class Library that contains xaml and xbf files to the output directory of the winforms app)

What do I need to do to make it work?

**Comment by niels9001** [comment]
@karmeye could you try removin the HeaderIcon, Description binding and/or Button.Content just to see what causes the crash?

**Reply by karmeye** [reply]
I removed all attributes, just kept `<ct:SettingsCard></ct:SettingsCard>`. I still get the same exception. No details.

![Screenshot 1](https://github.com/CommunityToolkit/Windows/assets/366038/56e5a687-98dc-4c83-a745-3824421ddb9b)

**Reply by niels9001** [reply]
What version of the SettingsControls package are you on? Do you have a standalone repro project we can take a look at?

**Reply by karmeye** [reply]
The latest: 8.0.240109

I have created a VS Solution that demonstrates the problem: https://github.com/karmeye/WinUI-WinForms-Issues

Build and run the WinForms project and click the Open WinUI Window button. As you can see it loads (even though you cannot enter text in the text box or close the window without the application closing, but that's another [issue](https://github.com/microsoft/WindowsAppSDK/discussions/4203#discussioncomment-8555041)). Then uncomment the XAML in `LibraryWindow.xaml` that contains the `SettingsCard`. No run and do the same thing again. (You might need to clean the solution for the changes to take effect, I don't know why).

Thx for your assistance

**Reply by michael-hawker** [reply]
Wonder if any other control works vs. just Segmented. Wonder if this is a general library XAML resource loading issue or not?

**Reply by karmeye** [reply]
@michael-hawker Yes, I replaced the `SettingsCard` with a `TokenizingTextBox` and get the same result.

> Wonder if this is a general library XAML resource loading issue or not?

Yes, could be, I get this error often, for example here: https://github.com/microsoft/microsoft-ui-xaml/issues/7163

However, the fact that in this case it works when using controls from WindowsAppSDK and not when adding controls from the CommunityToolkit prompted me to ask here.

**Reply by michael-hawker** [reply]
@karmeye the platform controls get/load their resources differently from third-party libraries. Have you tried any _other_ libraries controls as well? I'm going to suspect this isn't specific to the Toolkit. FYI @jeffstall, any thoughts?

**Reply by karmeye** [reply]
Only standard controls from WinAppSdk, which worked.

This issue is no longer relevant for me, it's ok by me if you want to close it.
Thanks.
